src/interpreter.js

```diff
@@ -40,26 +40,27 @@ import {
 // I know, I know... But this is not meant to be nice or final code. Just a quick rough approach to get a better feeling on the AST shape.
 
 const compileSentence = (sentence) => {
-  if(sentence.nodeType === 'VariableDeclaration') {
-    const { variable, writeable, value } = sentence
-    return ` ${writeable ? 'let' : 'const'} ${variable.name} = ${compileExpression(value)} ;`
-  }
-  if(sentence.nodeType === 'Assignment') {
-    const { variable, value } = sentence
-    return ` ${variable.name} = ${compileExpression(value)} ;`
+  switch (sentence.nodeType) {
```

Comment:
I look this up and the braces syntax defines a different scope for each case, so there is no scope problem :) The shitty thing is that the break seems to still be necessary if you don't return, which I think may be a bit misleading... But hey, at least we have scopes.